logstash-core/api/lib/app/system/info_command.rb

```diff
@@ -4,20 +4,30 @@
 class LogStash::Api::SystemInfoCommand < LogStash::Api::Command
 
   def run
-    report = { "version"   => "1.0.0",
-               "host_name" => "foobar",
-               "plugins"   => { "count" => 10, "list" => [] }
+    plugins_list = available_plugins
+    {
+      "version"   => LOGSTASH_CORE_VERSION,
+      "hostname" => hostname,
+      "plugins"   => { "count" => plugins_list.count, "list" => plugins_list.sort }
     }
-    report
   end
 
   private
 
-  def type
-    ["input", "output", "filter"][rand(3)]
+  def hostname
+    `hostname`.strip
   end
 
-  def name
-    ["elasticsearch", "json", "yaml", "translate"][rand(4)]
+  def available_plugins
+    all_available_plugins.map do |spec|
+      spec.name
+    end
   end
+
+  def all_available_plugins
+    Gem::Specification.find_all.select do |spec|
+      spec.metadata && spec.metadata["logstash_plugin"] == "true"
+    end
+  end
+
```

Comment:
I think we should do it, but since we want to iterate quickly I am +1 for a specific PR only with that feature? Lets create an issue and do it in a specific PR.